contracts/HubPool.sol

```diff
@@ -15,29 +19,38 @@ interface WETH9Like {
 }
 
 contract HubPool is Testable, Lockable, MultiCaller, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
     struct LPToken {
-        address lpToken;
-        bool isWeth;
+        ExpandedERC20 lpToken;
         bool isEnabled;
     }
 
+    WETH9Like public l1Weth;
+
     mapping(address => LPToken) public lpTokens; // Mapping of L1TokenAddress to the associated LPToken.
 
-    constructor(address timerAddress) Testable(timerAddress) {}
+    event LiquidityAdded(address l1Token, uint256 amount, uint256 lpTokensMinted, address indexed liquidityProvider);
+    event LiquidityRemoved(uint256 amount, uint256 lpTokensBurnt, address indexed liquidityProvider);
+
+    constructor(address _l1Weth, address _timerAddress) Testable(_timerAddress) {
+        l1Weth = WETH9Like(_l1Weth);
+    }
 
     /*************************************************
      *                ADMIN FUNCTIONS                *
      *************************************************/
 
     // TODO: the two functions below should be called by the Admin contract.
-    function enableL1TokenForLiquidityProvision(
-        address l1Token,
-        bool isWeth,
-        string memory lpTokenName,
-        string memory lpTokenSymbol
-    ) public onlyOwner {
-        ERC20 lpToken = new ERC20(lpTokenName, lpTokenSymbol);
-        lpTokens[l1Token] = LPToken({ lpToken: address(lpToken), isWeth: isWeth, isEnabled: true });
+    function enableL1TokenForLiquidityProvision(address l1Token) public onlyOwner {
+        ExpandedERC20 lpToken = new ExpandedERC20(
+            append("Across ", IERC20Metadata(l1Token).name(), " LP Token"), // LP Token Name
+            append("Av2-", IERC20Metadata(l1Token).symbol(), "-LP"), // LP Token Symbol
+            IERC20Metadata(l1Token).decimals() // LP Token Decimals
+        );
```

Comment:
Minor note: we could use a contract like the TokenFactory here down the line so the ERC20 bytecode doesn't need to live in this contract (and thus, we give ourselves more room in the bytecode limit).

Reply:
this is a good point. we can make this change if we run out of space. I'll add a comment.